### PR TITLE
Yeoman article

### LINKED ISSUE #17: Yeoman article
Noel's booked until later in April when he expects to write this article.

### PR DESCRIPTION
**Pull request by noelatfalafel**
Fixes #17

### CODE REVIEW

File: docs/client-side/yeoman.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+	- `Client-Side Build Support`_
+	- `Restoring, Building and Hosting from the Command Line`_
+	- `Adding to Your Project with Sub Generators`_
+	
+Getting Started
+---------------
+The `ASPNET generator <https://www.npmjs.com/package/generator-aspnet>`_ creates ASP.NET 5, DNX projects that can 
+be loaded into Visual Studio 2015 or run from the command line. The generator creates the following project types:
+  
+- Empty Application: An application with minimal dependencies that outputs "hello world" to the browser.
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: An empty Web application with minimal dependencies.

File: docs/client-side/yeoman.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+	- `Restoring, Building and Hosting from the Command Line`_
+	- `Adding to Your Project with Sub Generators`_
+	
+Getting Started
+---------------
+The `ASPNET generator <https://www.npmjs.com/package/generator-aspnet>`_ creates ASP.NET 5, DNX projects that can 
+be loaded into Visual Studio 2015 or run from the command line. The generator creates the following project types:
+  
+- Empty Application: An application with minimal dependencies that outputs "hello world" to the browser.     
+- Console Application: Another minimal application that outputs "hello world" to the console.
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: A DNX-based console application.

File: docs/client-side/yeoman.rst [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+- Web Application: A complete MVC web application with a simple home page and examples for managing accounts and login.  
+- Web API Application: An ASP.NET 5 web service with one controller.
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: A Web API built on MVC.

File: docs/client-side/yeoman.rst [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+Getting Started
+---------------
+The `ASPNET generator <https://www.npmjs.com/package/generator-aspnet>`_ creates ASP.NET 5, DNX projects that can 
+be loaded into Visual Studio 2015 or run from the command line. The generator creates the following project types:
+  
+- Empty Application: An application with minimal dependencies that outputs "hello world" to the browser.     
+- Console Application: Another minimal application that outputs "hello world" to the console.
+- Web Application: A complete MVC web application with a simple home page and examples for managing accounts and login.  
+- Web API Application: An ASP.NET 5 web service with one controller.   
+- Nancy ASP.NET Application: A light-weight HTTP service with one module.
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]:
```
A light-weight HTTP service with one module built using `Nancy <nancyfx.org>`_
```

File: docs/client-side/yeoman.rst [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+- Class Library: Creates a DLL that can be referenced by other ASP.NET projects.
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: A DNX-based class library.

File: docs/client-side/yeoman.rst [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+5.	Run the ``yo`` command and pass the name of the generator.
+
+.. code-block:: console 
+
+ yo aspnet
+
+6.	The aspnet generator displays a menu. Select the **Web Application** and press ``Enter``.
+
+.. image:: yeoman/_static/yeoman-yo-aspnet.png
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: Looks like the background coloring in PowerShell doesn't play nice with Yeoman. Can we get a new image from a simple cmd prompt?

File: docs/client-side/yeoman.rst [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+Client-Side Build Support
+-------------------------
+
+The `ASPNET generator <https://www.npmjs.com/package/generator-aspnet>`_ creates support files to configure client-side build tools. `Grunt or Gulp task runners <http://docs.asp.net/en/latest/client-side/grunt-gulp.html>`_ files are added to your project to automate build tasks for Web Application projects. By default, aspnet-generator creates **gruntfile.js** to run tasks. Running the generator with the **--gulp** argument generates **gulpfile.js** instead.
+
+.. code-block:: console 
+
+ yo aspnet --gulp
+ 
+The generator also configures **package.json** to load Grunt or Gulp and configures **project.json** to restore client-side packages
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: > and configures **project.json** to restore client-side packages using

and adds bower.json to  restore client-side packages using

File: docs/client-side/yeoman.rst [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+Building and Running from Visual Studio
+---------------------------------------
+
+You can load your generated ASP.NET 5 web project directly into Visual Studio 2015, then build and run your project from there.
+
+1. Open Visual Studio 2015. From the File menu select :menuselection:`Open --> Project/Solution`.
+
+2. In the Open Project dialog, navigate to the ``project.json`` file, select it and click the **Open** button. In the Solution Explorer, the project should look something like the screenshot below.
+
+ .. image:: yeoman/_static/yeoman-solution.png
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: this screen shot shows the package restore in progress (i.e. Restoring . . .). Can we instead have an image with package restore already completed?

File: docs/client-side/yeoman.rst [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+ 
+ .. image:: yeoman/_static/yeoman-home-page.png 
+ 
+Restoring, Building and Hosting from the Command Line
+-----------------------------------------------------
+
+You can prepare and host your web application using commands **dnu** (Microsoft .NET Development Utility) and 
+**dnx** (Micorosft .NET Execution Environment). 
+
+.. note:: For more information on DNX see `DNX Overview <http://docs.asp.net/en/latest/dnx/overview.html>`_
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]:
```
:doc:`\dnx\overview`
```

File: docs/client-side/yeoman.rst [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+5. Open a web browser and navigate to http://localhost:5000. 
+
+ .. image:: yeoman/_static/yeoman-home-page_5000.png 
+
+.. note:: You can also run the cross-platform Kestrel development server using the **dnx . kestrel** command. By default, Kestrel listens on port 5001 as defined in the **project.json**, **commands** section.  
+
+Adding to Your Project with Sub Generators
+------------------------------------------
+You can add new generated files using Yeoman even after the project is created. Use `sub generators <https://www.npmjs.com/package/generator-aspnet#sub-generatorsl>`_ to add any of the file types
```
Comment (danroth27) [inline docs/client-side/yeoman.rst]: Bad link - should be https://www.npmjs.com/package/generator-aspnet#sub-generators

**Comment by danroth27** [comment]
Looks good! Some minor editorial issues, but the content and structure looks solid :watch:

**Comment by ardalis** [comment]
@danroth27 ship it ?

**Comment by danroth27** [comment]
Looks good. Please squash your commits before your merge. :shipit:

**Comment by ardalis** [comment]
Merged in a separate PR.

**Pull request closed without merging**